sbtplugin/src/main/scala/scala/scalanative/sbtplugin/ScalaNativePlugin.scala

```diff
@@ -21,9 +21,15 @@ object ScalaNativePlugin extends AutoPlugin {
     val nativeEmitDependencyGraphPath = settingKey[Option[File]](
       "If non-empty, emit linker graph to the given file path.")
 
-    val nativeLinkAs = settingKey[Map[String, String]](
+    val nativeLibraryLinkage = settingKey[Map[String, String]](
       "Given a native library, provide the linkage kind (static or dynamic). " +
       "If key is not present in the map, dynamic is picked as a default.")
+
+    val nativeLink = inputKey[Unit](
+      "Generates native binary without running it.")
```

Comment:
It would be more idiomatic for `nativeLink` to return the `File` it produced. So basically it should be a `taskKey[File]`. Then `run` should actually *use* `nativeLink.value` as the path to the executable, rather than asking separately for `nativeBinary`.

To be even more idiomatic, it should be `taskKey[Attributed[Filed]]`. You might not realize yet, but eventually you'll need the attributes for one reason or another (we did).